Editor/EditorCore/DrawShapeTool.cs

```diff
@@ -238,60 +239,79 @@ internal void SetBounds(Vector3 size)
             m_BB_HeightCorner = m_BB_Origin + size;
         }
 
-        internal void DuplicatePreview(Vector3 position)
+        internal void DoDuplicateShapePreviewHandle(Vector3 position)
         {
-            if(position.Equals(Vector3.positiveInfinity) || !Event.current.isMouse)
-                return;
-
-            var pivotLocation = (PivotLocation)s_LastPivotLocation.value;
-            var size = currentShapeInOverlay.size;
+            var evt = Event.current;
 
-            m_Bounds.size = size;
+            bool previewShortcutActive = evt.shift && !(evt.control || evt.command);
 
-            Vector3 cornerPosition;
-            switch(pivotLocation)
+            if (evt.type == EventType.Layout)
             {
-                case PivotLocation.FirstCorner:
-                    cornerPosition = GetPoint(position);
-                    m_PlaneRotation = Quaternion.LookRotation(m_PlaneForward,m_Plane.normal);
-                    m_Bounds.center = cornerPosition + m_PlaneRotation * size / 2f;
-
-                    m_BB_Origin = cornerPosition;
-                    m_BB_HeightCorner = m_Bounds.center + m_PlaneRotation * (size / 2f);
-                    m_BB_OppositeCorner = m_BB_HeightCorner - m_PlaneRotation * new Vector3(0, size.y, 0);
-                    break;
-
-                case PivotLocation.Center:
-                default:
-                    position = GetPoint(position);
-                    cornerPosition = position - size / 2f;
-                    cornerPosition.y = position.y;
-                    m_Bounds.center = cornerPosition + new Vector3(size.x/2f,0, size.z/2f) + (size.y / 2f) * m_Plane.normal;
-                    m_PlaneRotation = Quaternion.LookRotation(m_PlaneForward,m_Plane.normal);
-
-                    m_BB_Origin = m_Bounds.center - m_PlaneRotation * (size / 2f);
-                    m_BB_HeightCorner = m_Bounds.center + m_PlaneRotation * (size / 2f);
-                    m_BB_OppositeCorner = m_BB_HeightCorner - m_PlaneRotation * new Vector3(0, size.y, 0);
-                    break;
-            }
+                if (HandleUtility.nearestControl == m_ControlID && previewShortcutActive)
```

Comment:
Not sure how true nearestControl will be depending on when this is called. If it's the end of the OnGUI it will be, otherwise a control that is closer might not have been added yet.

Reply:
this only matters on repaint, so it will be valid

Reply:
no wait you're right, fixing